internal/collector/processor/queued/queued_processor_test.go

```diff
@@ -16,16 +16,77 @@ package queued
 
 import (
 	"context"
+	"errors"
 	"sync"
 	"sync/atomic"
 	"testing"
-
-	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"time"
 
 	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+
+	"github.com/open-telemetry/opentelemetry-service/consumer"
 	"github.com/open-telemetry/opentelemetry-service/data"
+	"github.com/open-telemetry/opentelemetry-service/errorkind"
 )
 
+func TestQueuedProcessor_noEnqueueOnPermanentError(t *testing.T) {
+	ctx := context.Background()
+	td := data.TraceData{
+		Spans: make([]*tracepb.Span, 7),
+	}
+
+	c := &waitGroupTraceConsumer{
+		consumeTraceDataError: errorkind.Permanent(errors.New("bad data")),
+	}
+
+	qp := NewQueuedSpanProcessor(
+		c,
+		Options.WithRetryOnProcessingFailures(true),
+		Options.WithBackoffDelay(time.Hour),
+		Options.WithNumWorkers(1),
+		Options.WithQueueSize(2),
+	).(*queuedSpanProcessor)
+
+	c.Add(1)
+	if err := qp.ConsumeTraceData(ctx, td); err != nil {
+		// This is asynchronous so it should just enqueue, no errors expected.
+		t.Fatalf("c.ConsumeTraceData() = %v want nil", err)
+	}
+	c.Wait()
+	<-time.After(50 * time.Millisecond)
+
+	queueSize := qp.queue.Size()
+	if queueSize != 0 {
+		t.Fatalf("queueSize = %d, want 0", queueSize)
+	}
+
+	c.consumeTraceDataError = errors.New("transient error")
+	c.Add(1)
+	if err := qp.ConsumeTraceData(ctx, td); err != nil {
+		// This is asynchronous so it should just enqueue, no errors expected.
+		t.Fatalf("c.ConsumeTraceData() got non-permanent error")
+	}
+	c.Wait()
+	<-time.After(50 * time.Millisecond)
+
+	queueSize = qp.queue.Size()
```

Comment:
You can replace these 4 lines by single require.Equal() call. Similarly elsewhere. Makes tests much shorter.

Reply:
Need to get used to them :) I will add them